Sources/Lexer.swift

```diff
@@ -151,4 +162,21 @@ extension String {
     let last = findLastNot(character: character) ?? endIndex
     return self[first..<last]
   }
+  
+  func lineAndPosition(at range: Range<String.Index>) -> (content: String, number: Int, offset: String.IndexDistance) {
```

Comment:
I'd move this over to the reporter too, and internally Stencil would use character(or possibly byte?)-level source maps everywhere.